chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -554,39 +554,58 @@ void BraveRenderViewContextMenu::ExecuteAIChatCommand(int command) {
       ai_chat::features::kAIChatSSE.Get() && IsRewriteCommand(command) &&
       !source_web_contents_->GetUserData(kAIChatRewriteDataKey);
 
-  if (!rewrite_in_place) {
+  auto [action_type, p3a_action] = GetActionTypeAndP3A(command);
+  auto selected_text = base::UTF16ToUTF8(params_.selection_text);
+
+  if (rewrite_in_place) {
+    source_web_contents_->SetUserData(kAIChatRewriteDataKey,
+                                      std::make_unique<AIChatRewriteData>());
+    if (!ai_engine_) {
+      ai_engine_ = ai_chat::AIChatServiceFactory::GetForBrowserContext(
+                       source_web_contents_->GetBrowserContext())
+                       ->GetDefaultAIEngine();
+    }
+    ai_engine_->GenerateRewriteSuggestion(
+        selected_text, ai_chat::GetActionTypeQuestion(action_type),
+        ai_chat::BindParseRewriteReceivedData(
+            base::BindRepeating(&OnRewriteSuggestionDataReceived,
+                                source_web_contents_->GetWeakPtr())),
+        base::BindOnce(&OnRewriteSuggestionCompleted,
+                       source_web_contents_->GetWeakPtr(), selected_text,
+                       action_type));
+  } else {
+    auto* browser = GetBrowser();
+    if (!browser) {
+      VLOG(1) << "Can't get browser";
```

Comment:
this kind of logging should be removed before merge

Reply:
this is not new